rclcpp/test/rclcpp/CMakeLists.txt

```diff
@@ -669,13 +663,10 @@ if(TARGET test_interface_traits)
   target_link_libraries(test_interface_traits ${PROJECT_NAME})
 endif()
 
-# TODO(brawner) remove when destroying Node for Connext is resolved. See:
-# https://github.com/ros2/rclcpp/issues/1250
 ament_add_gtest(
   test_executors
   executors/test_executors.cpp
-  APPEND_LIBRARY_DIRS "${append_library_dirs}"
-  TIMEOUT 180)
```

Comment:
So it looks like this test on Windows is taking longer than 60 seconds.  So I think we need to put this back to 180 seconds.

Reply:
> 1. Is this the case for just this test, or does the other one need to be reverted too?

It looks like just for this `test_executors`, you can check the jobs above.

> 2. Is the problem only on windows?

It looks like it.  It's not really that surprising, Windows is much slower than Linux in all respects.

> 3. Do we need to fix the reasons why this test might be going wrong in relation to Connect?

Actually, this test isn't run on a per-RMW basis, only with the default one.  So the comment is bogus either way.

All of that said, it might be worthwhile taking a look at that particular test and seeing why it is taking over minute to complete.  If we can cut that down to something like 30 seconds, then we can probably lower this (and make CI faster to boot).

Reply:
I'm almost positive I found the problem [here](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/executors/test_executors.cpp#L637). It seems that the test that's taking forever is `testRaceConditionAddNode` taking about 20000 - 27000 ms, everything else compared to Linux, is just longer on Windows. 

This implementation uses `std::thread::hardware_concurrency`, so based on the number of cores that each system has, [cppreference here](https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency). It's the only place this is used in that test, other than in `multi_threaded_executor`'s test, but that's not an issue. My concern is how this works with the CI, since this is an actual hardware thing, can we find the number of cores we use, or is this just a Windows Software problem?